app/pb_kits/playbook/pb_card/_card.jsx

```diff
@@ -36,13 +79,30 @@ const Card = ({
     [`highlight_${highlight.color}`]: highlight.color,
   })
 
+  // coerce to array
+  const cardChildren = typeof(children) === 'object' && children.length ? children : [children]
+
+  const subComponentTags = (tagName) => {
+    return cardChildren.filter((c) => {
+      return c.type && c.type.displayName === tagName
+    }).map((child, i) => {
+      return React.cloneElement(child, { key: `${tagName.toLowerCase()}-${i}` })
+    })
+  }
+
+  const nonHeaderChildren = cardChildren.filter((child) => !child.type || child.type.displayName !== 'Header')
+
   return (
     <div className={classnames(cardCss, className)}>
+      {subComponentTags('Header')}
```

Comment:
What if we pass the actual reference to the component so we don't have to rely on `type.display_name` string comparisons?

```suggestion
      {subComponentTags(Header)}
```

Then type checking would simply be:

```js
const subComponentTags = Type => cardChildren.filter(c => c.type === Type)
                                             .map((child, i) => React.cloneElement(child, { key: `${Type.displayName.toLowerCase()}-${i}` }))

const nonHeaderChildren = cardChildren.filter(child => child.type !== Header)
```

Reply:
I implemented your suggested and the card header wouldn't render. I think we need to rely on string comparisons for React subcomponents, as all the other examples I found online uses string comparison as well.